# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: dom/canvas/WebGL2ContextSync.cpp
# Commit: ccfa767dba64
# Full Hash: ccfa767dba644dc193e0246eb8e8ff3377e8b8a5
# Author: Jeff Gilbert <jgilbert@mozilla.com>
# Date: 2020-01-09 03:40:18
# Regressor Bug: 1477756
# File Overlap Count: 1
# Description:
#   Bug 1477756 - Client-side bindings mirror for precise CC, and merge similar codepaths. r=handyman
#   
#   * Context loss using RAII
#   * Move Program reflection Client-side
#   
# ==============================================================================

diff -r 71c122ac0ca7 -r ccfa767dba64 dom/canvas/WebGL2ContextSync.cpp
--- a/dom/canvas/WebGL2ContextSync.cpp	Wed Jan 08 22:19:14 2020 +0000
+++ b/dom/canvas/WebGL2ContextSync.cpp	Wed Jan 08 22:19:16 2020 +0000
@@ -14,8 +14,7 @@
 // -------------------------------------------------------------------------
 // Sync objects
 
-already_AddRefed<WebGLSync> WebGL2Context::FenceSync(GLenum condition,
-                                                     GLbitfield flags) {
+RefPtr<WebGLSync> WebGL2Context::FenceSync(GLenum condition, GLbitfield flags) {
   const FuncScope funcScope(*this, "fenceSync");
   if (IsContextLost()) return nullptr;
 
@@ -34,19 +33,7 @@
   const auto& availRunnable = EnsureAvailabilityRunnable();
   availRunnable->mSyncs.push_back(globj);
 
-  return globj.forget();
-}
-
-bool WebGL2Context::IsSync(const WebGLSync* const sync) {
-  const FuncScope funcScope(*this, "isSync");
-  return ValidateIsObject(sync);
-}
-
-void WebGL2Context::DeleteSync(WebGLSync* sync) {
-  const FuncScope funcScope(*this, "deleteSync");
-  if (!ValidateDeleteObject(sync)) return;
-
-  sync->RequestDelete();
+  return globj;
 }
 
 GLenum WebGL2Context::ClientWaitSync(const WebGLSync& sync, GLbitfield flags,
@@ -87,59 +74,4 @@
   return ret;
 }
 
-void WebGL2Context::WaitSync(const WebGLSync& sync, GLbitfield flags,
-                             GLint64 timeout) {
-  const FuncScope funcScope(*this, "waitSync");
-  if (IsContextLost()) return;
-
-  if (!ValidateObject("sync", sync)) return;
-
-  if (flags != 0) {
-    ErrorInvalidValue("`flags` must be 0.");
-    return;
-  }
-
-  if (timeout != -1) {
-    ErrorInvalidValue("`timeout` must be TIMEOUT_IGNORED.");
-    return;
-  }
-
-  gl->fWaitSync(sync.mGLName, flags, LOCAL_GL_TIMEOUT_IGNORED);
-}
-
-MaybeWebGLVariant WebGL2Context::GetSyncParameter(const WebGLSync& sync,
-                                                  GLenum pname) {
-  const FuncScope funcScope(*this, "getSyncParameter");
-  if (IsContextLost()) return Nothing();
-
-  if (!ValidateObject("sync", sync)) return Nothing();
-
-  ////
-
-  const bool canBeAvailable =
-      (sync.mCanBeAvailable || StaticPrefs::webgl_allow_immediate_queries());
-  if (!canBeAvailable && pname == LOCAL_GL_SYNC_STATUS) {
-    return AsSomeVariant(LOCAL_GL_UNSIGNALED);
-  }
-
-  GLint result = 0;
-  switch (pname) {
-    case LOCAL_GL_OBJECT_TYPE:
-    case LOCAL_GL_SYNC_STATUS:
-    case LOCAL_GL_SYNC_CONDITION:
-    case LOCAL_GL_SYNC_FLAGS:
-      gl->fGetSynciv(sync.mGLName, pname, 1, nullptr, &result);
-
-      if (pname == LOCAL_GL_SYNC_STATUS && result == LOCAL_GL_SIGNALED) {
-        sync.MarkSignaled();
-      }
-
-      return AsSomeVariant(result);
-
-    default:
-      ErrorInvalidEnumInfo("pname", pname);
-      return Nothing();
-  }
-}
-
 }  // namespace mozilla